ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java

```diff
@@ -146,6 +148,27 @@ public void close() {
       segment.closeView();
     }
   }
+
+  /**
+   * Do data doPrefetch: from disk to memory buffer.
+   * @param relativeOffset the relativeOffset to start.
+   * @param size The size requested to doPrefetch.
+   */
+  void doPrefetch(long relativeOffset, long size) throws IOException {
+    long sizeToRead = Math.min(size, getMessageInfo().getSize() - relativeOffset);
+    prefetchedData = ByteBuffer.allocate((int) sizeToRead);
+    getChannel().position(offset.getOffset() + relativeOffset);
```

Comment:
I'm not sure this will work in a concurrent environment. Isn't this a shared channel? Can you instead use [this](https://docs.oracle.com/javase/8/docs/api/java/nio/channels/FileChannel.html#read-java.nio.ByteBuffer-long-)?

Reply:
Thanks!